CarCleaningService.rb

```diff
@@ -0,0 +1,145 @@
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+          end
+        @worker1.push(car)
+        return
+      elsif @worker2.empty?
+        car.car_time = car.start_time
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+        end
+        @worker2.push(car)
+        return
+      end
+      
+      #when a worker is busy
+      if @worker1.last.pickup_time < car.start_time
+        car.car_time = car.start_time
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+        end
+        @worker1.push(car)
+        return
+      end
+      if @worker2.last.pickup_time < car.start_time
+        car.car_time = car.start_time
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+        end
+        @worker2.push(car)
+        return
+      end
+      
+      #when both are busy
+      if @worker1.last.pickup_time < @worker2.last.pickup_time
+        car.car_time = @worker1.last.pickup_time 
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+        end
+        @worker1.push(car)
+        return
+      else
+        car.car_time = @worker2.last.pickup_time
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+        end
+        @worker2.push(car)
+      end
+    end
+      
+      #when we add a car we want to set the pickup time and to print it
+      def add_car
+        new_car = Car.new
+        pickup(new_car)
+        new_car.pickup_time.strftime("%A %d-%m-%Y %H:%M")
+      end
+
```

Comment:
this newline is a little bit unnecessary. not a big deal, but rubocop will complain about this